contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,807 @@
+         */
+        public int release() {
+            return INSTANCE.IOObjectRelease(this);
+        }
+    }
+
+    /**
+     * An IOKit iterator handle.
+     */
+    class IOIterator extends IOObject {
+        public IOIterator() {
+            super();
+        }
+
+        public IOIterator(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link IOKit#IOIteratorNext} on this object.
+         *
+         * @return If the iterator handle is valid, the next element in the iteration is
+         *         returned, otherwise {@code null} is returned. The element should be
+         *         released by the caller when it is finished.
+         */
+        public IORegistryEntry next() {
+            return INSTANCE.IOIteratorNext(this);
+        }
+    }
+
+    /**
+     * The base class for all objects in the registry.
+     */
+    class IORegistryEntry extends IOObject {
+        public IORegistryEntry() {
+            super();
+        }
+
+        public IORegistryEntry(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetRegistryEntryID} to return
+         * an ID for this registry entry that is global to all tasks.
+         *
+         * @param id
+         *            The resulting ID.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getRegistryEntryID(LongByReference id) {
+            return INSTANCE.IORegistryEntryGetRegistryEntryID(this, id);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetName} to return a name
+         * assigned to this registry entry.
+         *
+         * @param name
+         *            The caller's buffer to receive the name. This must be a 128-byte
+         *            buffer.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getName(Pointer name) {
+            return INSTANCE.IORegistryEntryGetName(this, name);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetChildIterator} to return an
+         * iterator over this registry entry’s child entries in a plane.
+         *
+         * @param plane
+         *            The name of an existing registry plane. Plane names are defined in
+         *            {@code IOKitKeys.h}, for example, {@code kIOServicePlane}.
+         * @param iter
+         *            The created iterator over the children of the entry, on success.
+         *            The iterator must be released when the iteration is finished.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getChildIterator(String plane, PointerByReference iter) {
+            return INSTANCE.IORegistryEntryGetChildIterator(this, plane, iter);
```

Comment:
Same comment as in `getRegistryEntryID`. In this case an `IOIterator` could be returned directly.